docs/sphinx/user_guide/feature/tiling.rst

```diff
@@ -0,0 +1,98 @@
+.. ##
+
+.. _tiling-label:
+
+===========
+Tiling
+===========
+
+In this section we discuss RAJA tiling statements. Many scientific
+computation algorithms can be classified as "Tiling" algorithms. A Tiling
+algorithm is one in which an iteration space is first broken up into a 
+collection of "tiles" and nested loops are used to iterate over tiles and 
+entries in a tile. This type of pattern is advantageous as operating on 
+smaller local chunks of data can improve performance via cache blocking. 
+For example, a for loop with a range of [0, 10)::
+
+  for(int i=0; i<10; ++i) {
+    //body
+  }
+
+May equivalently be expressed as loops which iterate over five tiles of 
+dimension two::
+
+      int numTiles = 5;
+      int tileDim  = 2;
+      for(int t=0; t<numTiles; ++t) {
+
+        for(int j=0; j<tileDim; ++j) {
+
+          //Calculation of global index
+          int i = j + tileDim*t;
+        }
+      }
+
+Tiling may be expressed in RAJA through kernel and tiling statements.
+As a starting point, we provide a RAJA variant of the tiled C++ loops above::
+
+    using KERNEL_EXEC_POL =
+    RAJA::KernelPolicy<
+      RAJA::statement::Tile<0, RAJA::statement::tile_fixed<2>, RAJA::seq_exec,
+        RAJA::statement::For<0, RAJA::seq_exec,
+                             RAJA::statement::Lambda<0>
+        >
+      >
+    >;
+
+   RAJA::kernel<KERNEL_EXEC_POL>(RAJA::make_tuple(RAJA::RangeSegment(0,10)), [=] (int i) {
+
+                         //body
+
+   });
+
+To tile an iteration space, a developer must use ``Tile`` and ``For`` statements templated on the same
+tuple range. The ``Tile`` statement is similar to a ``For`` statement but with the 
+added template argument of tile dimension. The ``Tile`` statement will iterate over the necessary tiles
+while the ``For`` statement will iterate over the entries in the tile. Together, the Tile and For statements will 
+return the global index within bounds of the original iteration space. Next, we discuss variations on tile 
+and for statements which provide the tile id number and iteration within the tile.
+
+The statement ``TileTCount`` is used to extract the tile number, while the statement ``ForICount``
+can be used to extract the iteration within the tile. TileTcount may be paired with a For statement,
+and ForICount may be paired with the Tile statement. Unlike the Tile statement, return values 
+for TileTCount and ForICount are returned in a kernel_param tuple and require specifying which tuple
+entry should hold the iterate. For example, the following kernel policy will populate the tile number in
+the first parameter entry and iteration within the tile in the second tuple entry. Parameter arguments 
+follow segment arguments in lambda arguments and are returned after the global index. ::
+
+  using KERNEL_EXEC_POL2 =
+    RAJA::KernelPolicy<
+      RAJA::statement::TileTCount<0, RAJA::statement::Param<0>, RAJA::statement::tile_fixed<2>, RAJA::seq_exec,

```

Comment:
Make it clear in the text above that the ``Param<#>`` statements are used to refer to which param tuple entry the 'for' statement corresponds to.

Also, note that fact that the global index always appears as the first lambda arg. Use a 'note' block for this.